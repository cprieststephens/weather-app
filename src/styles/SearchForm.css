.search-form button {
  background-color: #00003c;
  color: #fff;
  border: none;
  border-radius: 5%;
  padding: 0.4em;
  font-family: 'Raleway', sans-serif;
  font-weight: bold;
}

.search-form button:hover {
  background-color: #56ffac;
  color: #00003c;
  cursor: pointer;
}

.search-form input {
  padding: 0.25em;
}