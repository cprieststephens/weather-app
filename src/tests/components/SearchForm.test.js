import { render } from "@testing-library/react";
import SearchForm from "../../components/SearchForm";

describe("SearchForm", () => {
  const validProps = {
    setSearchText: jest.fn(),
    handleCitySearch: jest.fn(),
  };

  it("renders correctly", () => {
    const { asFragment } = render(
      <SearchForm
        searchText="Manchester"
        setSearchText={validProps.setSearchText}
        onSubmit={validProps.handleCitySearch}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the correct text in the button element", () => {
    const { getByText } = render(
      <SearchForm
        searchText="Manchester"
        setSearchText={validProps.setSearchText}
        onSubmit={validProps.handleCitySearch}
      />
    );
    expect(getByText("Search")).toBeInstanceOf(HTMLButtonElement);
  });
});
