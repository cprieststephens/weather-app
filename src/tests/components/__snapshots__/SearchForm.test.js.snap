// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm renders correctly 1`] = `
<DocumentFragment>
  <div
    class="search-form"
  >
    <input
      placeholder="Manchester"
      type="search"
      value="Manchester"
    />
    <button
      type="submit"
    >
      Search
    </button>
  </div>
</DocumentFragment>
`;
